core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -181,8 +196,112 @@ impl SchedulerController {
         }
     }
 
+    /// Forward packets to the next leader.
+    fn forward_packets(&mut self, hold: bool) {
+        const MAX_FORWARDING_DURATION: Duration = Duration::from_millis(100);
+        let start = Instant::now();
+        let bank = self.bank_forks.read().unwrap().working_bank();
+        let feature_set = &bank.feature_set;
+        let mut forwardable_packets =
+            ForwardPacketBatchesByAccounts::new_with_default_batch_limits();
+
+        // Pop from the container in chunks, filter using bank checks, then attempt to forward.
+        // This doubles as a way to clean the queue as well as forwarding transactions.
+        const CHUNK_SIZE: usize = 64;
+        let mut num_forwarded: usize = 0;
+        let mut ids_to_add_back = Vec::new();
+        let mut max_time_reached = false;
+        while !self.container.is_empty() {
+            let mut filter_array = [true; CHUNK_SIZE];
+            let mut ids = Vec::with_capacity(CHUNK_SIZE);
+            let mut txs = Vec::with_capacity(CHUNK_SIZE);
+
+            for _ in 0..CHUNK_SIZE {
+                if let Some(id) = self.container.pop() {
+                    ids.push(id);
+                } else {
+                    break;
+                }
+            }
+            let chunk_size = ids.len();
+            ids.iter().for_each(|id| {
+                let transaction = self.container.get_transaction_ttl(&id.id).unwrap();
+                txs.push(&transaction.transaction);
```

Comment:
can we not do this in the loop above where we fill `ids`?

Reply:
No. Popping ids is mutating change on container. this is getting reference to something in the container